pages/api/waitlist/mail.ts

```diff
@@ -0,0 +1,3 @@
+// mail here ??
+// write something similar to pages/api/mail/index.ts?
+// and call when someone unregisters for a class?
```

Comment:
@MatoPlus was wondering if you think this is the correct directory to write the POST(?) request? then the endpoint would be api/waitlist/mail 

i see the "Class in X hours" email in pages/api/mail/index.ts but also that doesn't seem to be called anywhere (is it run automatically?) ; i was thinking of writing the request here but the main difference is that it'd have to be called (after the deleteClassRegistration)

Reply:
`/api/mail/index` is a hook that is called via a hourly cron job via our AWS lambda (serverless) function. This lambda function is actually in the production account if you want to take a look. The PR was here https://github.com/uwblueprint/social-diversity-for-children/pull/69

I was thinking after a deleteRegistration api request is called, we just directly call a mail promise push in the api logic to send mails in the backend.

Similarly to what is in the `/api/mail/index` page,  we should add something similar to the following lines:
```typescript
        (parentRegs as any[]).forEach((reg) => {
            mailerPromises.push(
                send(
                    process.env.EMAIL_FROM,
                    reg.parent.user.email,
                    getEmailSubject(reg.parent.preferredLanguage),
                    classStartingSoonTemplate(
                        intervalHours,
                        imageLink,
                        name,
                        weekday,
                        startDate,
                        endDate,
                        startTimeMinutes,
                        durationMinutes,
                        reg.parent.preferredLanguage,
                    ),
                ),
            );
        });
```
^^ /api/mail line 90